### ISSUE TITLE
How can I help?

### ISSUE THREAD

**Issue by lukeapage** [jshint/jshint#2458]
- There are 55 PR's, many valid, many without any response comments
 - There are lots of open issues, some are questions, some are unreproducible, some are duplicates

What can I do to help? I'm not as experienced in jshint or ES6 as at least @caitp / @jugglinmike so I don't want to overstep the mark or do something you've decided not to do.

Having lots of open PR's puts off contributors. What can I do? do some need review? do some need closing? Are some waiting for jshint 3? Is there a plan for the next release - do you try and only include a few small PR's each time? there are quite a few small PR's from @jugglinmike and @nicolo-ribaudo that could be safe if reviewed okay - am I okay to pull them? (I understand you usually rebase each feature into a single commit which follows the contribution guidelines)

Having lots of open issues leads to duplicates and its hard to follow or work out what is valuable to fix. Would you mind if I go on a labeling spree? Are there some guidelines as to P1-P4 ? Do you mind if I close issues without clear reproduction cases or that appear dead? They can always be re-opened.

Does the proposal flag mean it is a un-ratified feature request ? or just that it is a feature request ? Should there be another label for feature request (vs bug?)?

**Comment by lukeapage** [comment]
something like this?

P1 - Warnings/Errors on valid JavaScript in common scenarios, including ES6 features. Otherwise popular or very important issues.
P2 - Warnings/Errors on valid JavaScript where code or feature is less common. Problems with overrides that can be worked around.
P3 - Missing warnings and quirks in behaviour that can be worked around.
P4 - Small and unimportant issues, with workarounds, which we don't think should effect anyone.

**Comment by jugglinmike** [comment]
> Are some waiting for jshint 3? Is there a plan for the next release - do you try and only include a few small PR's each time?

The ideas that we're deferring for JSHint 3 are marked with the "JSHint 3" label and filed under the "v3.0.0" milestone. Those markers don't signify any formal commitment, though. We've previously discussed timing version 3.0 to roughly coincide with the release of ES6, so we'll start talking about what makes the cut soon.

> there are quite a few small PR's from @jugglinmike and @nicolo-ribaudo that could be safe if reviewed okay - am I okay to pull them? (I understand you usually rebase each feature into a single commit which follows the contribution guidelines)

If there aren't any objections or outstanding requests-for-changes, then you should feel free to merge them (provided, of course, that they meet your expectations for quality/consistency and they don't introduce breaking changes).

> Do you mind if I close issues without clear reproduction cases or that appear dead? They can always be re-opened.

By all means! Just make that clear to each contributor as you are doing so.

> Does the proposal flag mean it is a un-ratified feature request ? or just that it is a feature request ? Should there be another label for feature request (vs bug?)?

We don't have a ratification process, so "proposal" and "feature request" are one in the same.

> something like this?

The readme has a looser interpretation, but I don't think it's worth splitting hairs over the boundaries. Generally speaking, assigning a label of any priority is a good way to validate new issues. If your framework produces a good distribution over 1-4, then I'd say, "go for it". We can always update the readme with your criteria later.

We definitely could have done a better job bringing you up to speed, so sorry for the confusion.

**Comment by jugglinmike** [comment]
Also @lukeapage, could you join the #jshint Freenode IRC channel?

**Comment by lukeapage** [comment]
@jugglinmike Would you be okay with @nicolo-ribaudo reviewing my PR's? I'm conscious there are quite alot now that are needing review. @nicolo-ribaudo would you be happy doing that?

**Comment by jugglinmike** [comment]
Nope. It's important that @jshint/core maintains some level of awareness of the state of `master`. We're appreciate your efforts; we just don't have the bandwidth that you have right now. For my part, I'm hoping to catch up a bit this weekend.
